File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchyVB/RadGridViewEFCodeFirst.Data/Common/DataGenerator.vb
Imports RadGridViewEFCodeFirst.Models

Public Class DataGenerator
    Public Shared Sub PopulateData(data As IRadGridViewEFCodeFirstData)
        For i As Integer = 1 To 100
            Dim orderType As New OrderType() With {
                 .OrderTypeId = i,
                 .Description = "Test" & i
            }

            Dim order As New Order() With {
                 .OrderId = i, _
                 .Description = "Description" & i,
                 .OrderTypeId = orderType.OrderTypeId
            }

            Dim shipper As New Shipper() With {
                 .ShipperId = i,
                 .Name = "Name" & i,
                 .OrderTypeId = orderType.OrderTypeId,
                 .Address = "Address" & i
            }

            data.OrderTypes.Add(orderType)
            data.Orders.Add(order)
            data.Shippers.Add(shipper)

            If i Mod 10 = 0 Then
                data.SaveChanges()
            End If
        Next

        data.SaveChanges()
    End Sub
End Class

File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchy/RadGridViewEFCodeFirst.Data/Common/DataGenerator.cs
using RadGridViewEFCodeFirst.Data.Contracts;
using RadGridViewEFCodeFirst.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadGridViewEFCodeFirst.Data.Common
{
    public class DataGenerator
    {
        public static void PopulateData(IRadGridViewEFCodeFirstData data) 
        {
            for (int i = 1; i <= 100; i++)
            {
                OrderType orderType = new OrderType()
                {
                    OrderTypeId = i,
                    Description = "Test" + i
                };

                Order order = new Order()
                {
                    OrderId = i,
                    Description = "Description" + i,
                    OrderTypeId = orderType.OrderTypeId
                };

                Shipper shipper = new Shipper()
                {
                    ShipperId = i,
                    Name = "Name " + i,
                    OrderTypeId = orderType.OrderTypeId,
                    Address = "Address " + i,
                };

                data.OrderTypes.Add(orderType);
                data.Orders.Add(order);
                data.Shippers.Add(shipper);

                if (i % 100 == 0)
                {
                    data.SaveChanges();
                }
            }

            data.SaveChanges();
        }
    }
}
